client.go

```diff
@@ -154,12 +162,18 @@ func (c *Client) run() {
 		waiters  = map[uint32]recvRequest{}
 		queued   = map[uint32]received{} // messages unmatched by waiter
 		incoming = make(chan received)
+		shutdown = make(chan struct{})
 	)
 
 	go func() {
+		close(shutdown)
```

Comment:
This shutdown is not doing anything right now, it should probably be deferred for future use or removed